packages/hash/frontend/src/pages/shared/ory-kratos.ts

```diff
@@ -72,8 +72,10 @@ export const createFlowErrorHandler =
           return;
         }
         case "session_already_available":
-          // User is already signed in, let's redirect them home!
-          await router.push("/");
+          // User is already signed in, if we're in the login flow let's redirect them home!
+          if (flowType === "login") {
+            await router.push("/");
+          }
           return;
         case "session_refresh_required": {
           // We need to re-authenticate to perform this action
```

Comment:
(comment for line 130)
In development I get `404`s sometimes when I'm on the login page and the flow id is invalidated somehow. Could we add a case for `404` as well next to `410` to also just redirect to the flow type?

Reply:
Good idea. I think this happens when you re-seed the database, because the flow is nowhere to be found. This isn't something we'd therefore have to handle in production, but we can handle it in development. Have added this in https://github.com/hashintel/hash/pull/950/commits/b6e4e04b886669bfddf6e7b6b5c69d01d13fed17.

Regarding the `404` error, aren't we already doing this on line 140?

Reply:
Not sure i follow, what do you mean we already do this on line 140 for `404`? 

Also a little unsure if b6e4e04b886669bfddf6e7b6b5c69d01d13fed17 does fix this, I think it'd be safer to `await router.push(`/${flowType}`);` before returning. When testing locally, an inactive `flowId` fails to login with this update

Reply:
> what do you mean we already do this on line 140 for 404?

Apologise, I was referring to your request to address the `410` error - which we are already handling on line 140 of the file.

> an inactive flowId fails to login with this update

Do you mean an inactive existing flow? Or a non-existent flow?

Reply:
I've modified the switch case to reset the flow if there is a 404 error in development: https://github.com/hashintel/hash/pull/950/commits/ad1ef0e2ce0d013546e66ec58465000efd6108f7

This will only trigger for non-existent flows though, not expired ("inactive") flows. These should be handled by the `self_service_flow_expired` switch case.

Reply:
> Do you mean an inactive existing flow? Or a non-existent flow?
Non-existent flow.

This new fix addresses the error with a non-existent flow (after re-seeding DB as you pointed out) :)